### DISCUSSION TITLE
magicksave should support bitdepth parameter

### DISCUSSION THREAD

**Discussion by proyb6** [libvips/libvips#2015] [Ideas]
Compiled with the libvips master branch and latest ImageImagick with module enabled, Safari web browser could not support JPEG2000 in 16-bit sRGB.

Original image
Identify x.jpg JPEG 960x372 960x372+0+0 8-bit sRGB 72072B 0.000u 0:00.000

vips magicksave x.jpg x.jp2 --format JP2
Identify x.jp2 JP2 960x372 960x372+0+0 16-bit sRGB 420564B 0.000u 0:00.000

From Imagemagick convert x.jpg x.jp2
Identify x.jp2 JP2 960x372 960x372+0+0 8-bit sRGB 373349B 0.000u 0:00.000

**Comment by jcupitt** [comment]
Yes, it looks like there's some other setting that needs to be exposed. A Q value would be useful too.

**Comment by proyb6** [comment]
I like to suggest a parameter,

depth=8
depth=16
depth=32

I assume using IM Q16 libraries would be slower than Q8, so I have recompiled with IM Q8, if it's possible for VIPS to link to any available Q# in /usr/local/lib would be awesome.

**Comment by jcupitt** [comment]
Unfortunately, when you configure imagemagick with QN=xx it changes the size of various types, so you have to also rebuild everything that uses libMagick. You can't flip a switch at runtime.

**Comment by proyb6** [comment]
I see, that worked for me by recompile with 8-bit libMagick source. Thanks!

**Comment by jcupitt** [comment]
It should be possible for libvips to save as 8-bit JP2 using an imagemagick compiled at Q16. I'll reopen for investigation.

**Comment by jcupitt** [comment]
8.10 has a new `bitdepth` parameter for savers. magicksave should use `bitdepth=8` to write an 8-bit image rather then the default 16.

Let's tag this as an enhacement, perhaps for 8.11.
